### ISSUE TITLE
Can you migrate this to Libraries 2.0

### ISSUE THREAD

**Issue by ScruffR** [tomdeboer/SparkCoreDallasTemperature#10]
Currently this lib does not build with 2.0 projects anymore due to the `../` includes.

If you can't or don't want to, can we take ownership of this library for Particle Elite?

**Comment by tomdeboer** [comment]
#12 Should have fixed it.

**Comment by ScruffR** [comment]
Nope, libraries 2.0 requires a different file structure too (as commented on #12)

**Comment by ScruffR** [comment]
I've filed  PR for this
https://github.com/tomdeboer/SparkCoreDallasTemperature/pull/13

Please merge and reupload to Particle Library repository

**Comment by wrshannon** [comment]
any progress on resolving this?

**Comment by GregSawin** [comment]
Can we fork this and do what's necessary to get this into Particle's 2.0 format?  The DS18B20 library (in particle) is lacking many features of this library.

**Comment by tomdeboer** [comment]
Yeah definitely, just fork it and make the changes you need :)

**Comment by GregSawin** [comment]
@ScruffR did you want to do that?  
I've never pushed a particle library before but if that's what it'll take I'll learn ;-)

**Comment by ScruffR** [comment]
Particle needs to take ownership of that library on their side to allow anybody else to upload an updated version.
Without that only the original contributor can do that.

The issue with "stale"/abandoned libraries is a reoccuring topic when we talk to Particle :wink: 
I think @bsatrom is now in charge of libraries.

> DS18B20 library (in particle) is lacking many features of this library

Can you list some?
It may be possible to add some.

**Comment by GregSawin** [comment]
Alarms is the big one that's missing but I don't personally use that.

The main feature I like is that you can request the temp from all sensors at once.  This can either block until all are ready or return immediately and let you query for the status.  After that you can quickly grab the temp from each sensor by index.  You can also get a total device count as well as get an address by index.  

While neither library does this, I think the library should handle the temp validation and automatically retry as needed.  The relative flakiness of the dallas sensors seems to be a real pain point for people and neither library provides a good out of the box way to get and confirm a good read.  In other words I'd like to be able to suggest a library and have the example work robustly as is and not have to tell them how to alter the code to deal with the bad reads.  Perhaps the library could internally track bad reads and provide a function to check this data to help identify wiring issues or a failing sensor.

So personally I'm using a mix of both libraries as well as custom versions I've created from them.  I'd like to consolidate all my programs to use the same library so I've recently started looking into the pros & cons of each.  If I can get close enough with a standard Web IDE compatible library then I'd use that and likely submit my enhancements via github PR so everyone can benefit.

**Comment by bsatrom** [comment]
Thanks @ScruffR I've added this to the list and we'll work on getting it ported over.
